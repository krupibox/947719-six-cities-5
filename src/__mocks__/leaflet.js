const Leaflet = jest.genMockFromModule(`leaflet`);

Leaflet.icon = () => {};

Leaflet.map = () => {
  return {
    setView: () => {},
  };
};

Leaflet.marker = () => {
  return {
    addTo: () => {},
  };
};

Leaflet.tileLayer = () => {
  return {
    addTo: () => {},
  };
};

Leaflet.layerGroup = () => {
  return {
    addTo: () => {},
  };
};

module.exports = Leaflet;
