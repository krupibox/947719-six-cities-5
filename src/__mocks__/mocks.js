export const TestMock = {
  activeCity: `Cologne`,

  cities: [`Paris`, `Brussels`, `Hamburg`, `Amsterdam`, `Cologne`],

  firstCity: [`Cologne`],

  review: {
    user: {
      id: 1,
      avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg`,
      name: `Max`,
    },
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    date: `2020-11-05T09:55:08.500Z`,
    rating: 4.1,
  },

  reviews: [{
    user: {
      id: 1,
      avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg`,
      name: `Max`,
    },
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    date: `2020-11-05T09:55:08.500Z`,
    rating: 4.1,
  }],

  favorites: {
    Cologne: [{
      id: 1,
      city: {
        name: `Paris`,
        location: {
          latitude: 0,
          longitude: 0,
        }
      },
      previewImage: `img/apartment-01.jpg`,
      images: [
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-03.jpg`,
      ],
      type: `Apartment`,
      price: 80,
      rating: 3.1,
      title: `Beautiful &amp; luxurious apartment at great location`,
      coordinates: [52.3909553943508, 4.85309666406198],
      host: {
        id: 1,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`
      },
      description: `Discover daily local life in city center.`,
      isPremium: true,
      location: {
        latitude: 0,
        longitude: 0,
      }
    }],
    Hamburg: [{
      id: 1,
      city: {
        name: `Paris`,
        location: {
          latitude: 0,
          longitude: 0,
        }
      },
      previewImage: `img/apartment-01.jpg`,
      images: [
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-03.jpg`,
      ],
      type: `Apartment`,
      price: 80,
      rating: 3.1,
      title: `Beautiful &amp; luxurious apartment at great location`,
      coordinates: [52.3909553943508, 4.85309666406198],
      host: {
        id: 1,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`
      },
      description: `Discover daily local life in city center.`,
      isPremium: true,
      location: {
        latitude: 0,
        longitude: 0,
      }
    }],
  },

  offer: {
    id: 1,
    city: {
      name: `Paris`,
      location: {
        latitude: 0,
        longitude: 0,
      }
    },
    previewImage: `img/apartment-01.jpg`,
    images: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
    ],
    type: `Apartment`,
    price: 80,
    rating: 3.1,
    title: `Beautiful &amp; luxurious apartment at great location`,
    coordinates: [52.3909553943508, 4.85309666406198],
    host: {
      id: 1,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Discover daily local life in city center.`,
    isPremium: true,
    isFavorite: true,
    location: {
      latitude: 0,
      longitude: 0,
    }
  },

  offers: [
    {
      id: 1,
      city: {
        name: `Paris`,
        location: {
          latitude: 0,
          longitude: 0,
        }
      },
      previewImage: `img/apartment-01.jpg`,
      images: [
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-03.jpg`,
      ],
      type: `Apartment`,
      price: 80,
      rating: 3.1,
      title: `Beautiful &amp; luxurious apartment at great location`,
      coordinates: [52.3909553943508, 4.85309666406198],
      host: {
        id: 1,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`
      },
      description: `Discover daily local life in city center.`,
      isPremium: true,
      isFavorite: true,
      location: {
        latitude: 0,
        longitude: 0,
      }
    }
  ],

  cityCoords: {
    cityName: `Cologne`,
    latitude: 0,
    longitude: 0,
    zoom: 13
  },

  offersCoords: [{
    offerId: 1,
    latitude: 0,
    longitude: 0,
    zoom: 13
  }],

  info: {
    // eslint-disable-next-line camelcase
    avatar_url: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
    email: `Oliver.conner@gmail.com`,
    id: 1,
    // eslint-disable-next-line camelcase
    is_pro: false,
    name: `12345678`,
  },
};
