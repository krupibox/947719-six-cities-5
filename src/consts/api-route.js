export const APIRoute = {
  HOTELS: `/hotels`,
  REVIEWS: `/comments`,
  FAVORITE: `/favorite`,
  LOGIN: `/login`,
};
