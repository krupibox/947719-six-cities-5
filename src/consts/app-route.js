const AppRoute = {
  LOGIN: `/login`,
  RESULT: `/result`,
  ROOT: `/`,
};

export default AppRoute;
