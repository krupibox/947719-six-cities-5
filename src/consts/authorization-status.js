const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export default AuthorizationStatus;
