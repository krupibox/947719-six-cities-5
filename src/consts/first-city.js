export const FIRST_CITY = 0;

