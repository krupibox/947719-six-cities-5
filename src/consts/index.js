export {AuthorizationStatus} from './authorization-status';
export {APIRoute} from './api-route';
export {AppRoute} from './app-route';
