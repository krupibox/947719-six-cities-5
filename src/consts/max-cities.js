const MAX_CITIES = 6;

export default MAX_CITIES;
