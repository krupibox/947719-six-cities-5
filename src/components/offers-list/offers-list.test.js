import {Router} from 'react-router-dom';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import browserHistory from "../../browser-history";
import OffersList from './offers-list';
import {AuthorizationStatus} from "../../consts/authorization-status";
import {TestMock} from '../../__mocks__/mocks';

const mockStore = configureStore([]);

const store = mockStore({
  USER: {
    authorizationStatus: AuthorizationStatus.NO_AUTH,
    authorizationInfo: {},
  }
});

describe(`<OfferList/>`, () => {
  it(`should render correctly`, () => {
    const tree = renderer
            .create(
                <Provider store={store}>
                  <Router
                    history={browserHistory}
                  >
                    <OffersList
                      offers={TestMock.offers}
                      nearby={false}
                      favorite={false}
                      onCardHover={() => {}}
                    />)
                  </Router>
                </Provider>)
            .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
