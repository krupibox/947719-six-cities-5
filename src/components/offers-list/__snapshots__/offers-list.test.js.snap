// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OfferList/> should render correctly 1`] = `
Array [
  <article
    className=" cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="false place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="img/apartment-01.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className="false place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            80
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <a
          className="place-card__bookmark-button button"
          href="/login"
          onClick={[Function]}
        >
          <svg
            className="place-card__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </a>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__name"
          href="/offer/1"
          onClick={[Function]}
        >
          Beautiful &amp; luxurious apartment at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>,
  ")",
]
`;
